src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,22 +620,32 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String[] linesArray, int duplicateEmailIndex) {
+        return "Same email address as the student in line \"" + linesArray[duplicateEmailIndex + 1] + "\"";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    private String addInvalidStudentInfo(StudentAttributes student, String sanitizedLine) {
```

Comment:
can you leave a header comment to describe what is `sanitizedLine`? It isn't intuitive on first look. You can refer to the caller method `createStudents` as there's a `@param lines`, which is similar to `sanitizedLine`.

Also, you should actually rename `sanitizedLine` because you can't tell what this variable refers to from its naming. `sanitizedLine` refers to something like the details of the students or something like that isn't it.